### PR TITLE
First implementation of list endpoint to retrieve all saved tenancies

### PR DESCRIPTION
**Pull request by elcorbs**
This endpoint is just returning everything in the `tenagree` table, joining with the two lookup tables to get tenure and agreement types.

Also renamed `TenancyReference` to `TenancyAgreementReference` in the domain object, even though it's more verbose I found it confusing with the shortened name.

### CODE REVIEW

File: TenancyInformationApi.Tests/V1/Gateways/TenancyGatewayTests.cs [hunk 1]
```diff
@@ -56,14 +56,41 @@ public void GetByIdWillOnlyGetLookupValuesForZAGTypeLookups()
...
+        [Test]
+        public void ListTenanciesWithNoSavedTenanciesReturnsAnEmptyList()
+        {
+            _classUnderTest.ListTenancies().Should().BeEmpty();
+        }
+
+        [Test]
+        public void ListTenanciesGivenNoQueryParametersWillReturnAllSavedTenancies()
+        {
+            var savedEntities = new List<(UhTenancyAgreement uhTenancy, UhTenureType tenureTypeLookup, UhAgreementType agreementTypeLookup)>
```
Comment (LBHBomotoso) [inline TenancyInformationApi.Tests/V1/Gateways/TenancyGatewayTests.cs:74]: Are you creating the list with the returned values from the function?

**Review by LBHBomotoso — COMMENTED** [review]
All good

**Review by LBHBomotoso — APPROVED** [review]

File: TenancyInformationApi.Tests/V1/Gateways/TenancyGatewayTests.cs (hunk 1 above)
Comment (elcorbs) [reply TenancyInformationApi.Tests/V1/Gateways/TenancyGatewayTests.cs:74]: Yeah, so a list of tuples.

Comment (LBHBomotoso) [reply TenancyInformationApi.Tests/V1/Gateways/TenancyGatewayTests.cs:74]: neat

**Pull request merged**